### PR TITLE
testing: mm test will be skipped to prevent memory overflow

### PR DESCRIPTION
**Pull request by chenzihan0416**
### Summary

In the mm test, if the memory requested this time exceeds 3/4 of the remaining memory,
the request will be skipped to avoid insufficient memory.

### **Impact**
testing/mm

### **Testing**
qemu/sim run mm

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



I need the content of your PR to determine if it meets the NuttX requirements. Please provide the following:

* **A link to the PR** or
* **The content of your PR description** 

With that information, I can help you assess whether your PR meets the necessary criteria.

**Comment by cederom** [comment]
Thanks @chenzihan0416. Please update the PR description, we are not accepting undescribed untested changes, see https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md.

File: testing/mm/mm_main.c [hunk 1]
```diff
@@ -119,11 +119,23 @@ static const int g_alignment2[NTEST_ALLOCS / 2] =
 
 static FAR void       *g_allocs[NTEST_ALLOCS];
 static struct mallinfo g_alloc_info;
+int allow_malloc_faill = 0;
```
Comment (xiaoxiang781216) [inline testing/mm/mm_main.c:122]: why need this flag, @chenzihan0416 ?

Comment (chenzihan0416) [reply testing/mm/mm_main.c]: This flag has been deleted and the first submission requires the maintainer's help to trigger CI, thank you! @xiaoxiang781216

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by cederom — APPROVED** [review]
Thank you @chenzihan0416 :-)

File: testing/mm/mm_main.c [hunk 2]
```diff
@@ -124,6 +124,17 @@ static struct mallinfo g_alloc_info;
...
+static bool is_oversize(int size)
+{
+  if (size < 0)
+    {
+      return 0;
+    }
+
+  unsigned long threshold = g_alloc_info.mxordblk * 3 / 4;
+  return threshold < size;
+}
```
Comment (XuNeo) [inline testing/mm/mm_main.c:136]: Let's be normal person.

```suggestion
static bool is_oversize(int size)
{
  if (size < 0)
    {
      return false;
    }

  unsigned long threshold = g_alloc_info.mxordblk * 3 / 4;
  return size > threshold;
}
```

Comment (XuNeo) [inline testing/mm/mm_main.c:136]: It feels more natural to return `size > threshold`.
Also return false instead of 0 please.

**Review by XuNeo — APPROVED** [review]

File: testing/mm/mm_main.c (hunk 2 above)
Comment (chenzihan0416) [reply testing/mm/mm_main.c]: Thanks. Modified according to your comments. @XuNeo

**Comment by cederom** [comment]
@chenzihan0416 please let us know when all is set for merge and final test build, are short on CI resources.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**